docs/rtk-query/usage/customizing-queries.mdx

```diff
@@ -939,7 +939,7 @@ const api = createApi({
       async queryFn(_arg, _queryApi, _extraOptions, fetchWithBQ) {
         // get a random user
         const randomResult = await fetchWithBQ('users/random')
-        if (randomResult.error) throw randomResult.error
+        if (randomResult.error) return { error: randomResult.error as FetchBaseQueryError } 
```

Comment:
No need to JS examples is needed, it will be generated from this. I'm not sure that the type assertion here is required though - are you sure about that?

Reply:
Gotcha! And thats a good question. Im actually not sure. I added it in because if `await fetchWithBQ(`user/${user.id}/posts`)` results in an error, then we assert the type of the error with `FetchBaseQueryError` . 

I also found this in this docs - https://redux-toolkit.js.org/rtk-query/usage-with-typescript#error-result-example. So im not actually sure 😅